Moodle 4.4 compatibility (with backwards compatibility)

Buda9 opened the pull request:
In Moodle 4.4 I was getting "PHP Notice: Callback before_standard_html_head in tool_cohortheader component should be migrated to new hook callback for core\hook\output\before_standard_head_html_generation" and similar, so I made a fix.

Could you please take a look when you're available?

Thank you

Buda9 committed:
Moodle 4.4 compatibility (with backwards compatibility)

danmarsden commented:
Thanks for the PR! - typically our policy is to branch instead of maintain backwards compatibility to avoid technical debt and our preference is to use git commit history as the changelog rather than maintaining a standalone changelog file.

If you get a chance, feel free to update the PR to remove the changelog file, and introduce a breaking change (not compatible with 4.3 or lower) and we'll just create a new branch for MOODLE 4.4. and higher and merge that change in at some point.

no worries if you don't want to do this - we'll just get round to manually merging this in at some point in future.

thanks!

Buda9 committed:
Breaking change: Moodle 4.4 (and up) comaptibility

- deleted CHANGELOG.md
- Restored original README.md
- version.php bump to 0.5 and Moodle 4.4 min required
- small fixes
- tested and everything works great

Buda9 commented:
Thank you for the prompt response @danmarsden. I've made changes so it's only 404 compatible

- deleted CHANGELOG.md
- Restored original README.md
- version.php bump to 0.5 and added Moodle 4.4 requirement
(if you don't want me to touch version file, I can revert it to original version)
- small fixes
- tested with 404 and everything works great
